### PR TITLE
chore(operations): Update Rusoto to 0.41.x

### LINKED ISSUE #1103: Update `rusoto` version
We are currently a few versions behind the current rusoto release. We should update to the latest version. This seems simple but they reworked their error's in `0.38` and thus we can't upgrade without changing a bunch of code. So this task is a bit more mechanical.

### PR DESCRIPTION
**Pull request by phyber**
Closes #1103 

This PR updates Rusoto to 0.41.x. The x is because the credentials crate is at 0.41.1, while the rest are at 0.41.0.

There was a single easily dealt with type change during this update.

### CODE REVIEW

File: src/sinks/aws_kinesis_streams.rs [hunk 1]
```diff
@@ -243,6 +244,8 @@ fn encode_event(
             .unwrap_or_default(),
     };
 
+    let data = Bytes::from(data);
```
Comment (LucioFranco) [inline src/sinks/aws_kinesis_streams.rs]: oh cool! they updated to use bytes! https://docs.rs/rusoto_kinesis/0.41.0/rusoto_kinesis/struct.PutRecordsRequestEntry.html#structfield.data woohoo!

File: src/sinks/aws_s3.rs [hunk 2]
```diff
@@ -642,7 +642,7 @@ mod integration_tests {
             Ok(_) | Err(RusotoError::Service(CreateBucketError::BucketAlreadyOwnedByYou(_))) => {}
             Err(e) => match e {
                 RusotoError::Unknown(b) => {
-                    let body = String::from_utf8(b.body.clone()).unwrap();
+                    let body = String::from_utf8(b.body.clone().to_vec()).unwrap();
```
Comment (LucioFranco) [inline src/sinks/aws_s3.rs]: pretty minor but we probably want to use `String::from_utf8_lossy(&b.body[..])` here.

**Comment by phyber** [comment]
Should be good to go!

File: src/sinks/elasticsearch.rs [hunk 3]
```diff
@@ -262,7 +262,7 @@ fn es(
                     // to play games here
                     let body = request.payload.take().unwrap();
                     match body {
-                        SignedRequestPayload::Buffer(body) => builder.body(body).unwrap(),
+                        SignedRequestPayload::Buffer(body) => builder.body(body.to_vec()).unwrap(),
```
Comment (LucioFranco) [inline src/sinks/elasticsearch.rs]: we should apply that same trick above here.

Comment (phyber) [reply src/sinks/elasticsearch.rs]: Spoke too soon, on it :)

Comment (phyber) [reply src/sinks/elasticsearch.rs]: It doesn't appear that the same trick works here.

**Comment by lukesteensen** [comment]
Thanks so much for this @phyber!

**Pull request merged**